sklearn/utils/validation.py

```diff
@@ -980,3 +980,56 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
 
     if max_val is not None and x > max_val:
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
+
+
+def _check_sample_weight(sample_weight, y=None, n_samples=None, dtype=None,
+                         order=None, **kwargs):
+    """Validate sample weights
+
+    Parameters
```

Comment:
We can introduce this in the next PR if we don't break any back-compatibility.

Reply:
Yes, I would rather make a separate PR for, it might also involve defining a global setting `require_positive_sample_weight` to ether 'all', 'any' or False cf related discussion in https://github.com/scikit-learn/scikit-learn/issues/12464#issuecomment-433815773